api/jobs-api/src/main/java/org/browsermob/core/har/HarNameValuePair.java

```diff
@@ -57,17 +58,27 @@ public void setValue(String value) {
     this.value = value;
   }
 
-  // TODO: Perhaps these should be done the right way
-  public boolean equals(Object o) {
-    HarNameValuePair obj = (HarNameValuePair) o;
+  @Override
+  public boolean equals(Object other) {
```

Comment:
Maybe it will be simpler to use https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/builder/EqualsBuilder.html